napari/_vispy/canvas.py

```diff
@@ -382,9 +386,10 @@ def _process_mouse_event(
             return
 
         # Add the view ray to the event
-        event.view_direction = self.viewer.camera.calculate_nd_view_direction(
-            self.viewer.dims.ndim, self.viewer.dims.displayed
-        )
+        # event.view_direction = self.viewer.camera.calculate_nd_view_direction(
+        #     self.viewer.dims.ndim, self.viewer.dims.displayed
+        # )
+        event.view_direction = self._calculate_view_direction(event.pos)
```

Comment:
I think this will need to be converted into nD world coordinates. I think as-is, this will be in the 3D coordinates of the rendered scene. (not totally sure - didn't look too closely)

Reply:
Yeah, I think this is the main remaining hurdle.

Basically, this code with a 4D image is currently breaking (testing by just moving mouse on the canvas with #7126 changes included):

```
import napari
from skimage.data import binary_blobs

blobs = binary_blobs(
    length=64,
    blob_size_fraction=0.2,
    n_dim=4,
    volume_fraction=0.1,
)
v = napari.Viewer(ndisplay=3)
v.add_image(blobs)
```

Essentially we need the 3D view direction to be embedded in an nD vector so it can be properly sliced later.